specs/WinRT/WinRTAPIContracts.md

```diff
@@ -0,0 +1,78 @@
+
+
+
+# Windows App SDK API Contracts
+
+## Summary
+Customers trust that Microsoft products are sound.  As Windows App SDK evolves and more developers use Windows App SDK in their own projects Windows App SDK needs a way, at build time and run time, to allow apps to light up API features and to gracefully degrade.  This can be achieved by looking at the deltas between APIs.  
+
+In order to allow apps to depend on API versions Windows App SDK will add contracts to all their WinRT APIs.
+
+
+## Architecture Decisions
+1. Contract boundaries
+   1. One contract per feature/component/sub-system
+   2. Contracts will not span across a transport package/repo.
+2. Contract numbering.
+Contracts will be versioned with one number, starting with 1 and is incremented for any interface change.  This version does not correlate with the Windows App SDK version.
+3. Enforcement
+The build systems will have enforcement gates to make sure that contracts are present, up to date, and there are no conflicts.
+4. Contract Migration.
+Contract Migration will not be a part of this spec.  Windows App SDK will add a spec on contract migration if, and when, there is a situation where it is needed.
+5. Internal contracts
+All WinRT internal APIs will have an contract and the version will always be 1 regardless of any interface changes.
```

Comment:
All WinRT types are contracted, not just interfaces (enum, runtimeclass, other types of types &lt;sic&gt; need contracts).